go/vt/tabletserver/tabletserver.go

```diff
@@ -176,6 +181,63 @@ func NewTabletServer(config Config) *TabletServer {
 	return tsv
 }
 
+// createTransactionThrottlerOrPanic creates the transaction throttler from the configuration
+// given in the relevant fields in tsv.config. It does nothing if the transaction throttler is not
+// enabled (tsv.config.EnableTxThrottler is false).
+// It panics if any error occurs or if the tabletserver already has a transaction throttler.
+func (tsv *TabletServer) createTransactionThrottlerOrPanic() {
+	if !tsv.config.EnableTxThrottler {
+		return
+	}
+	if tsv.transactionThrottler != nil {
+		panic("Transaction throttler already exists!")
+	}
+	var throttlerConfig throttlerdatapb.Configuration
+	err := proto.UnmarshalText(
+		tsv.config.TxThrottlerConfig,
+		&throttlerConfig)
+	if err != nil {
+		panic(fmt.Errorf("Can't parse TxThrottlerConfig: %v", err))
```

Comment:
Changed to disable the throttler on error. Add a TODO to return an error from NewTabletServer.